stk/calculators/optimization/macromodel.py

```diff
@@ -1068,15 +1083,65 @@ def generate_com(self, mol):
         logger.debug(f'Creating .com file for "{mol.name}".')
 
         # Define some short aliases to keep the following lines neat.
-        temp = self.temperature
-        sim_time = self.simulation_time
-        tstep = self.time_step
+        # And checking whether input satisfies requirements.
+        if self.temperature > 99999.99:
+            raise MacroModelInputError(
+                'Supplied temperature (> 99999 K) is too high.'
+            )
+
+        else:
+            temp = self.temperature
+
+        if self.conformers > 9999:
+            raise MacroModelInputError(
+                'Supplied number of conformers (> 9999) is too high.'
+            )
+
+        if self.simulation_time > 999999.99:
+            raise MacroModelInputError(
+                'Supplied simulation time (> 999999 ps) is too long.'
+            )
+
+        elif self.simulation_time > 99999.99:
+            sim_time = - self.simulation_time/100
+
+        else:
+            sim_time = self.simulation_time
+
+        if self.time_step > 99999.99:
+            raise MacroModelInputError(
+                'Supplied time step (> 99999 fs) is too high.'
+            )
+
+        else:
+            tstep = self.time_step
+
+        if self.eq_time > 999999.99:
+            raise MacroModelInputError(
+                'Supplied eq time (> 999999 ps) is too long.'
+            )
+
+        elif self.eq_time > 99999.99:
+            eq_time = - self.eq_time/100
+
+        else:
+            eq_time = self.eq_time
+
+        if self.minimum_gradient < 0.0001:
+            raise MacroModelInputError(
+                'Convergence gradient (< 0.0001) is too small.'
+            )
+
+        if self.maximum_iterations > 999999:
+            raise MacroModelInputError(
+                'Number of iterations (> 999999) is too high.'
+            )
```

Comment:
All of these checks should be made during `__init__`, not during `generate_com`.